specutils/core/generic.py

```diff
@@ -0,0 +1,119 @@
+from specutils.spectrum1d import Spectrum1D
+
+import numpy as np
+import logging
+
+from astropy.nddata import NDArithmeticMixin, NDSlicingMixin, NDIOMixin
+import astropy.units as u
+from astropy.wcs import WCS
+
+
+class GenericSpectrum1D(NDIOMixin, NDSlicingMixin, NDArithmeticMixin,
+                        Spectrum1D):
+    """
+    A generalized spectrum data object that implements several `NDData`
+    features such as IO, arithmetic, and slicing.
+    """
+
+    # The Spectrum1D wcs_attribute object is useless to us, let's avoid it
+    _wcs_attributes = {}
+
+    def __init__(self, data, name="", dispersion=None, dispersion_unit=None,
+                 uncertainty=None, wcs=None, *args, **kwargs):
+        super(GenericSpectrum1D, self).__init__(flux=data,
+                                                wcs=wcs,
+                                                uncertainty=uncertainty,
+                                                indexer="UNUSED",
+                                                *args, **kwargs)
+        self._dispersion = dispersion
+        self._dispersion_unit = dispersion_unit
+        self.name = name or "New Data Object"
+
+    def from_self(self, copy=True, **kwargs):
```

Comment:
This should be a class method. I think that in other parts of Astropy we would call this method ``clone``, or ``copy`` - @eteq, what do you think? I wonder if this could instead be ``def copy(self, deepcopy=True, **kwargs)`` and if ``deepcopy`` is ``False`` the arrays themselves are not copied?